### DISCUSSION TITLE
Android : Navigating from one page to another keeps the first one visible.

### DISCUSSION THREAD

**Discussion by mcNets** [unoplatform/uno#17433] [Q&A] [answered]
Using a NavigationView, when I navigate from one page to the next one, the first page is still visible. 

Background of the second page is set to Transparent because I apply Mica backdrop in the Windows head.

Is this related to https://github.com/unoplatform/uno/discussions/17277?

![Captura de pantalla 2024-07-09 174031](https://github.com/unoplatform/uno/assets/24267381/10f55c2c-b6b2-44cd-82a6-2bc9bd383ded) ![Captura de pantalla 2024-07-09 174322](https://github.com/unoplatform/uno/assets/24267381/2f329a93-072c-4437-9b4a-c52bca63166c)

**Comment by kazo0** [comment, marked as answer]
This is most probably due to the [NativeFramePresenter](https://github.com/unoplatform/uno.toolkit.ui/blob/main/src/Uno.Toolkit.UI/Controls/NavigationBar/NativeFramePresenter.Android.cs) logic that is coming from the use of the Uno Toolkit.

You can validate this by setting the [`FeatureConfiguration.NativeFramePresenter.AndroidUnloadInactivePages`](https://github.com/unoplatform/uno/blob/d67dab22600338570b3e541e7672277c45790e9a/src/Uno.UI/FeatureConfiguration.cs#L573-L582) feature flag to `true` in your App.xaml.cs like we are doing here: https://github.com/unoplatform/uno.toolkit.ui/blob/0ab54f3190188628e51264b895fd69c34a6de6fe/samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/App.xaml.cs#L75-L77

**Reply by mcNets** [reply]
@kazo0 Thanks Steve.

**Reply by jeromelaban** [reply]
Also, this is an expected behavior w.r.t the background of the page. If you don't set one, it will be transparent. The NativeFramePresenter keeps pages loaded for performance reasons.

**Reply by mcNets** [reply]
@jeromelaban are you suggesting that it is better to keep them in memory?

**Reply by jeromelaban** [reply]
In general, yes, it avoids reattaching the page's visual tree on back navigation.
